internal/endpoints/endpoints.go

```diff
@@ -47,12 +51,14 @@ func (ps Partitions) ResolveEndpoint(region string, opts Options) (aws.Endpoint,
 
 // Partition is an AWS partition description for a service and its' region endpoints.
 type Partition struct {
-	ID                string
-	RegionRegex       *regexp.Regexp
-	PartitionEndpoint string
-	IsRegionalized    bool
-	Defaults          Endpoint
-	Endpoints         Endpoints
+	ID                 string
+	RegionRegex        *regexp.Regexp
+	PartitionEndpoint  string
+	IsRegionalized     bool
+	Defaults           Endpoint
+	DualStackDefaults  Endpoint
+	Endpoints          Endpoints
+	DualStackEndpoints Endpoints
```

Comment:
Just curious if we could have updated the `Endpoint` to include an `isDualStack` identifier? 

That way, we can continue using `Defaults` and `Endpoints` member, and not need the `DualStackDefaults` and `DualStackEndpoints` member. 

Might help in future to avoid duplicate members in case another property similar to DualStack is added for an endpoint.

Reply:
This is constraint of the modeling choices for `endpoints.json` where the pseudo-region key in the `map[string]Endpoint` can overlap between dual-stack and non dual-stack.